workbook/serializers/sign_up_serializer.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+from django.core.exceptions import BadRequest
+from rest_framework import serializers
+from workbook.models.models import *
+
+
+class SignUpSerializer(serializers.ModelSerializer):
+    # choiceField returns default error message, and it does not allow you to easily customize the error message.
+    # CharField it does not have a predefined validation.
+    role = serializers.CharField()
+
+    class Meta:
+        model = User
+        fields = ['first_name', 'last_name', 'email', 'password', 'role']
+
+    # Serializer Validation/ Field-level validation
+    def validate_first_name(self, value):
+        if len(value) < 2:
+            raise serializers.ValidationError("Name must be at least 2 characters long.")
+        return value
+
+    def validate_last_name(self, value):
+        if len(value) < 2:
+            raise serializers.ValidationError("Name must be at least 2 characters long.")
```

Comment:
make the 2 a variable and import it so when it is to be changed no need to change it multiple times.